test/utils/deployment.go

```diff
@@ -244,3 +244,37 @@ func WaitForObservedDeployment(c clientset.Interface, ns, deploymentName string,
 		return c.Extensions().Deployments(ns).Get(deploymentName, metav1.GetOptions{})
 	}, desiredGeneration, 2*time.Second, 1*time.Minute)
 }
+
+// Pool until deployment status and its underlying resources reach the desired state.
+func WaitForDeploymentCompletes(c clientset.Interface, d *extensions.Deployment, logf LogfFn, pollInterval, pollTimeout time.Duration) error {
+	var (
+		deployment *extensions.Deployment
+		reason     string
+	)
+
+	err := wait.Poll(pollInterval, pollTimeout, func() (bool, error) {
+		var err error
+		deployment, err = c.Extensions().Deployments(d.Namespace).Get(d.Name, metav1.GetOptions{})
```

Comment:
Remember that `Extensions()` is deprecated, versions should be specified explicitly where possible.

We can use any API version that supports deployment. For now, using extensions/v1beta1 is fine, and we'll migrate these calls at some point altogether.

Reply:
yup!